extensions/wikia/Email/Controller/ForumController.class.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+use Email\Fatal;
+
+class ForumController extends EmailController {
+
+	protected $titleText;
+	protected $titleUrl;
+	protected $details;
+	/* @var \Title */
+	protected $board;
+
+	public function initEmail() {
+		$this->titleText = $this->request->getVal( 'titleText' );
+		$this->titleUrl = $this->request->getVal( 'titleUrl' );
+		$this->board = \Title::newFromText(
+			$this->request->getVal( 'boardTitle' ),
+			$this->request->getVal( 'boardNamespace' )
+		);
+		$this->details = $this->getVal( 'details' );
+
+		$this->assertValidParams();
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'details' => $this->getDetails(),
+			'detailsHeader' => $this->getDetailsHeader(),
+			'buttonText' => $this->getButtonText(),
+			'buttonLink' => $this->getButtonLink(),
+		] );
+	}
+
+	public function getSubject() {
+		return wfMessage( $this->getSubjectKey(), $this->boardText )
+			->inLanguage( $this->targetLang )
+			->text();
+	}
+
+	protected function getSubjectKey() {
+		return 'emailext-forum-subject';
```

Comment:
Is there a reason to have the `getSubjectKey()`, `getDetails()`, `getButtonLink()`, `getDetailsHeader()` methods? All they're doing is just returning the corresponding instance variable. Can we use those instance variable instead inside of the `body()` method?